### PR TITLE
Suppress warnings

### PR DESCRIPTION
**Pull request by okkez**
In previous version about 2500 lines of warnings.
In this version about 1200 lines of warnings.

Remained warinigs:

* From ServerEngine about 400 lines
  * See https://github.com/fluent/serverengine/pull/47
* `Encoding.default_external`, `Encoding.default_internal` about 800 lines
* Some other warnings about 50 lines

### CODE REVIEW

File: lib/fluent/plugin/output.rb [hunk 1]
```diff
@@ -561,8 +561,8 @@ def handle_stream_with_standard_format(tag, es)
           meta_and_data[meta].add(time, record)
           records += 1
         end
-        meta_and_data.each_pair do |meta, es|
```
Comment (tagomoris) [inline lib/fluent/plugin/output.rb]: Please remove these change: I'm fixing these on other branches.

File: lib/fluent/plugin_id.rb [hunk 2]
```diff
@@ -47,11 +47,15 @@ def plugin_id_for_test?
...
     def plugin_id_configured?
-      @_id_configured
+      if instance_variable_defined?("@_id_configured")
+        @_id_configured
+      end
     end
 
     def plugin_id
-      @id ? @id : "object:#{object_id.to_s(16)}"
+      if instance_variable_defined?("@id")
```
Comment (tagomoris) [inline lib/fluent/plugin_id.rb]: This method should return `"object:#{object_id.to_s(16)}"` if instance_variable `@id` is not defined.

**Comment by okkez** [comment]
I've pushed commits and rebased for all comments.

**Comment by tagomoris** [comment]
Great!

**Comment by okkez** [comment]
Thanks!

**Pull request merged**